File: Sources/RepoBarCore/Models/SourceControlProvider.swift
import Foundation

public enum SourceControlProvider: String, CaseIterable, Codable, Hashable, Sendable, Identifiable {
    case github
    case gitlab
    case bitbucketCloud
    case forgejo
    case gitea
    case customGit

    public var id: String {
        self.rawValue
    }

    public var label: String {
        switch self {
        case .github: "GitHub"
        case .gitlab: "GitLab"
        case .bitbucketCloud: "Bitbucket Cloud"
        case .forgejo: "Forgejo"
        case .gitea: "Gitea"
        case .customGit: "Custom Git"
        }
    }
}

File: Windows/RepoBar.Core/Models/SourceControlProvider.cs
namespace RepoBar.Core.Models;

public enum SourceControlProvider
{
    GitHub,
    GitLab,
    BitbucketCloud,
    Forgejo,
    Gitea,
    CustomGit,
}

public static class SourceControlProviderExtensions
{
    public static string Label(this SourceControlProvider provider) =>
        provider switch
        {
            SourceControlProvider.GitHub => "GitHub",
            SourceControlProvider.GitLab => "GitLab",
            SourceControlProvider.BitbucketCloud => "Bitbucket Cloud",
            SourceControlProvider.Forgejo => "Forgejo",
            SourceControlProvider.Gitea => "Gitea",
            SourceControlProvider.CustomGit => "Custom Git",
            _ => provider.ToString(),
        };
}
